### ISSUE TITLE
Create generator widget

### ISSUE THREAD

**Issue by jrmi** [jrmi/airboardgame#8]
A generator can wrap any other widget and add ability to clone a model infinity

**Comment by jrmi** [comment]
Is it necessary if the master can add any item ?

**Comment by dudefaceguy** [comment]
Generating items on the fly will help with performance, especially on mobile. In my games (Mattock and YIMBY), there are large stacks of identical items, which degrades performance. YIMBY, for example, has 216 tokens, but only 56 are used on the board in the initial setup. The rest are sitting there waiting for the players to use them, but only about 80 will actually be used in any given game. This is also true for Go, which has stacks of 360 identical pieces, none of which are on the board when the game starts. These games are very laggy on mobile, and removing the large stacks of identical pieces fixes the lag.

Adding an "infinite bag" tool, like in Tabletop Simulator, would be very helpful. I know that you can select an item, popup the menu, and hit the clone button, but this is not an intuitive replication of the tabletop experience. Also, if I drag the item onto the board, it is often difficult to select the exact item that I want to clone when it is jumbled up with the other tokens around it. So, I would like to see an infinite bag tool. Thanks again!

**Comment by jrmi** [comment]
You're right on utility of infinite bag tool and I put this ticket in next version roadmap.

However, I also noticed a performance issue after your message on item movement when there is many items in the game and I think I fix it. Can you try again on your mobile and tell me if it improve performances ?

Thanks for reporting.

**Comment by dudefaceguy** [comment]
Item movement seems smooth, but moving and resizing the screen is still very laggy. This is actually very important, since zooming is the first thing you do when starting a game, and you have to redo it every time you reload the page. Being able to set a default zoom level might help, or just making the default zoom much farther out on mobile, since most games seem to start zoomed in too far on mobile especially.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by jrmi** [comment]
I'm working on screen resize performance but I already change the default zoom when you load a game, what do you think ?

**Comment by jrmi** [comment]
Hey @dudefaceguy

I have added a new item: the generator that mimic the behaviour of infinite bag.

You can test it here https://616b3dd2ca5b34d6510e0d4d--airboardgame.netlify.app
